infra/gcp/clusters/projects/k8s-infra-public-pii/main.tf

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  project_id  = "k8s-infra-public-pii"
+  bucket_name = "k8s-infra-artifacts-gcslogs"
+  dataset-id  = replace(local.bucket_name, "-", "_")
+}
+
+
+data "google_billing_account" "account" {
+  billing_account = "018801-93540E-22A20E"
+}
+
+data "google_organization" "org" {
+  domain = "kubernetes.io"
+}
+
+
+resource "google_project" "project" {
+  name            = local.project_id
+  project_id      = local.project_id
+  org_id          = data.google_organization.org.org_id
+  billing_account = data.google_billing_account.account.id
+}
+
+resource "google_project_service" "project" {
+  project = google_project.project.id
+
+  for_each = toset([
+    "bigquery.googleapis.com",
+    "bigqueryreservation.googleapis.com",
+    "bigquerytransfer.googleapis.com",
+    "storage-component.googleapis.com"
+  ])
+
+  service = each.key
+}
+
+// Ensure k8s-infra-gcp-auditors@kubernetes.io can administer all the Biquery
+// resources except the datasets and the datasets tables
+resource "google_project_iam_member" "auditors-bq-resources-admins" {
+  project = google_project.project.id
+  role    = "roles/bigquery.resourceAdmin"
+  member  = "group:k8s-infra-gcp-auditors@kubernetes.io"
+}
+
+// BigQuery dataset for audit logs
+resource "google_bigquery_dataset" "audit-logs-gcs" {
+  dataset_id                  = local.dataset-id
+  friendly_name               = local.dataset-id
+  delete_contents_on_destroy  = false
+  default_table_expiration_ms = 34560000000 #30 days
+  location                    = "US"
+  project                     = google_project.project.project_id
+}
+
+
+// Ensure k8s-infra-gcp-auditors@kubernetes.io has admin access to the dataset
+resource "google_bigquery_dataset_iam_member" "auditors-bq-dataset-editors" {
+  dataset_id = local.dataset-id
+  role       = "roles/bigquery.dataEditor"
+  member     = "group:k8s-infra-gcp-auditors@kubernetes.io"
+}
+
+
+# A bucket to store logs in audit logs for GCS
+resource "google_storage_bucket" "audit-logs-gcs" {
+  name          = local.bucket_name
+  project       = google_project.project.project_id
+  storage_class = "REGIONAL"
+  location      = "us-central1"
+
+  lifecycle_rule {
+    action {
+      type = "Delete"
+    }
+
+    condition {
+      age = 400
+    }
+  }
+
+  // NOTE: Prevent the bucket from being deleted
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+// Grant write access to group-cloud-storage-analytics@google.com
+resource "google_storage_bucket_acl" "analytics_write" {
```

Comment:
Can we do this with IAM instead of ACLs?

Reply:
Done.